libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -1681,7 +1681,11 @@ bool AP_Arming::arm(AP_Arming::Method method, const bool do_arming_checks)
         }
     }
 #endif
-
+#if defined(HAL_ARM_GPIO_PIN)
+    if (!AP_BoardConfig::arming_gpio_disabled()) {
```

Comment:
add a static function that does this, update_arming_pin(), and call it from both arm() and disarm()

Reply:
done